### DISCUSSION TITLE
Does not block some elements on US bank site

### DISCUSSION THREAD

**Discussion by smtips01** [uBlockOrigin/uBlock-issues#3350] [Q&A]
### Prerequisites

- [X] I verified that this is not a filter list issue. Report any issues with filter lists or broken website functionality in the [uAssets issue tracker](https://github.com/uBlockOrigin/uAssets/issues).
- [X] This is **NOT** a [_YouTube_](https://www.github.com/uBlockOrigin/uAssets/issues/20586), [_Facebook_](https://www.github.com/uBlockOrigin/uAssets/issues/3367) or [_Twitch_](https://www.github.com/uBlockOrigin/uAssets/issues/5184) report. _These sites MUST be reported by clicking their **respective links**_.
- [X] This is not a support issue or a question. For support, questions, or help, visit [/r/uBlockOrigin](https://www.reddit.com/r/uBlockOrigin/).
- [X] I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=is%3Aissue) to avoid opening a duplicate issue.
- [X] The issue is not present after disabling uBO in the browser.
- [X] I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I am reporting is not normal behavior.

### I tried to reproduce the issue when...

- [X] uBO is the only extension.
- [X] uBO uses default lists and settings.
- [X] using a new, unmodified browser profile.

### Description

||onlinebanking.usbank.com/digital/servicing/customer-dashboard/assets/welcome-background-gradient-dk.jpg does not block at all. 

### A specific URL where the issue occurs.

```yaml
https://onlinebanking.usbank.com/digital/servicing/customer-dashboard
```

### Steps to Reproduce

Try blocking the element with the greeting at the top of the page

### Expected behavior

For the section to get blocked. Does not work.

### Actual behavior

Nothing but removing the overlay blue color, grey and letting all stay and repeated attempts to block do nothing

### uBO version

1.59.0

### Browser name and version

Firefox 129.0.1

### Operating System and version

Mac OS 14.6.1

**Comment by stephenhawk8054** [comment]
Refresh while bypassing the cache (`Command` + `Shift` + `R`) and test again.

**Comment by garry-ut99** [comment]
> smtips01:  **Actual behavior**: Nothing but removing the overlay blue color, grey and letting all stay and repeated attempts to block do nothing

If "Refresh while bypassing the cache" won't work, then try to create additional filter, because there is nothing more than the overlay blue color in your current filter: `onlinebanking.usbank.com/digital/servicing/customer-dashboard/assets/welcome-background-gradient-dk.jpg`
<br>
> smtips01: **Steps to Reproduce**: Try blocking the element with the greeting at the top of the page

Can't help more, because nothing like that appears for me, neither on Firefox, nor on Chromimum, nor on Mobile, and the URL redirects to: `https://www.usbank.com/index.html`, given it's a bank site and by looking at the URL, most likely the "greeting" appear only for logged in users with an account.

<hr>

If that happens to be "Refresh while bypassing the cache" issue, then OP's issue is invalid and is also a duplicate of :
- https://github.com/uBlockOrigin/uBlock-issues/issues/3299
- https://github.com/uBlockOrigin/uBlock-issues/issues/1107

>  * [x] (smtips01): I performed a [cursory search of the issue tracker](https://github.com/uBlockOrigin/uBlock-issues/issues?q=sort%3Aupdated-desc+is%3Aissue) to avoid opening a duplicate issue.
>  * [x] (smtips01): I checked the [documentation](https://github.com/gorhill/uBlock/wiki) to understand that the issue I am reporting is not normal behavior.

**Comment by smtips01** [comment]
Just to note, problem was never resolved but it's marked as such. Wow, like 7-8 months later :) No, bypassing cache didn't work, I can try to block and block over and over again, it refreshes the area blocked and literally does nothing. I don't appreciate someone saying 'resolved" when it never was. I get you apparently can't do further troubleshooting because it requires a bank login, but by no stretch should it be marked RESOLVED when it wasn't at all.

**Reply by stephenhawk8054** [reply]
If you block an image when it's already loaded, you will always need to clear the cache. There's nothing uBO can do. That's browser's feature. You didn't reply anything to us, what do you want us to input more?

If you don't want to experience that, when you use element picker, use cosmetic filters, don't use network filters. uBO displays other options for you to choose which filters to use.

That's all we can answer. If you still don't understand something, please post screenshot when you use element picker.

**Comment by smtips01** [comment]
Sounds like you're implying that I don't know how to clear the cache. I don't think anyone using your addon would NOT be savvy enough to know that. This ticket was opened way last August. I've probably changed computers 3 times since then and done completely fresh installs of Firefox and uBlock. Can't go blaming it on me and making it sound like I don't know what I'm doing. I already told you I understand you can't fully toubleshoot this because it's behind a bank login. I get it. 

To provide you with a bit more info that might help you deal with the addon just not working in certain instances, here's what I did just now.

Right click anywhere on the box to block, I get what's in the first screen shot. Choose Create to block what's selected. Nothing happens.  Does nothing. Multiple repeated attempts do nothing. 

Use the Picker. Choose an individual element in the box I want to block. It automatically de-clicks the item within the box when I actually click the item, then highlights the entire larger box background as in the first screen shot. It only highlights on mouse-over as indicated in the second screen shot. Because I use a Mac not Windows, I was able to take a screen shot with the highlighted item on mouse-over.

![Screenshot 2025-04-07 at 2 55 34 PM](https://github.com/user-attachments/assets/fcbdaca1-c5ec-4098-818e-6d447f054987)

![Screenshot 2025-04-07 at 2 56 03 PM](https://github.com/user-attachments/assets/b593b51b-5c75-4e6a-8c3f-665a6282e050)

Finally, it's really not something you need to take offense about or think that I'm an idiot who doesn't know what they are doing, I do know what I'm doing. I only posted my previous response, because too often, online support marks something as RESOLVED when it really wasn't, and that only confuses the next guy trying to get help. Nothing was resolved, and I accept that in this case, that it won't get resolved. The bank site is doing some weird layering that doesn't allow uBlock to block some elements on the site. 

Thanks for your time, have a good day.

**Reply by gorhill** [reply]
> marks something as RESOLVED

It's a discussion, it was marked as **answered**, after months of no feedback, resulting in a stale discussion. That's just normal housekeeping stuff.

If you find a page on that site which suffers the same issue without requiring logging in, we can look into it, otherwise we can only provide vague guidance when we are unable to reproduce on our side.

**Reply by stephenhawk8054** [reply]
I don't take offense of anything. What I mean is simply what I wrote:

> You didn't reply anything to us, what do you want us to input more?

If an issue doesn't have any replies to our comments, we will just mark it as resolved, because what else do we have to input, when we already comment to it: https://github.com/uBlockOrigin/uBlock-issues/discussions/3350#discussioncomment-10384549 ?

> Just to note, problem was never resolved but it's marked as such. Wow, like 7-8 months later :)

> by no stretch should it be marked RESOLVED when it wasn't at all.

> I don't appreciate someone saying 'resolved" when it never was.

How do we know that? We cannot read minds. I don't know why you think my comment was offensive when the reply was like that. It sounds to me like "we already know that the solution above doesn't work but we close it anyways"?

> online support

This is not online support. If you are expecting that, this is the wrong tool. We are volunteers of a community FOSS project, there's no "online supports" here. We help what we could based on the information of the issue. If there are no more information from the reporters of the issue after we already comment on it and the issue becomes stale, we close it. That's all.

---

> I already told you I understand you can't fully toubleshoot this because it's behind a bank login. I get it.

The problem is not about fully troubleshooting the issue. The problem is there's literally no comments after that for us to say anything more. Your issue just simply contains this filter:

```adb
||onlinebanking.usbank.com/digital/servicing/customer-dashboard/assets/welcome-background-gradient-dk.jpg
```

which is a network filter to block a `.jpg` image, and the resolution to network filters is this (that's why it's marked as resolved): https://github.com/uBlockOrigin/uBlock-issues/discussions/3350#discussioncomment-10384549 . That's all we can answer based on those few information. Your screenshots above do not look like an image, so it's not about the filter above any more, the issue is different now.

If possible, please post the full screenshots that have full uBO element picker's filter options like this:

![image](https://github.com/user-attachments/assets/fc5600d3-4d71-459b-91bf-09c442bd3058)
